content/blog/architecture-as-code-serverless/index.md

```diff
@@ -0,0 +1,170 @@
+			'Principal': {
+				'Service': 'ecs-tasks.amazonaws.com'
+			},
+			'Action': 'sts:AssumeRole',
+		}]
+	})
+)
+
+rpa = aws.iam.RolePolicyAttachment('task-exec-policy',
+	role=role.name,
+	policy_arn='arn:aws:iam::aws:policy/service-role/AmazonECSTaskExecutionRolePolicy'
+)
+```
+
+To learn more about creating [reusable components]({{< relref "/docs/intro/concepts/programming-model#components" >}}), read about Pulumi's programming model, which shows how to author components.
+
+## Polyglot applications
+
+This example shows how to deploy multiple functions in Google Cloud Functions. Note that we have one function written in Python and another in Go. The code that deploys the function is written in python and creates a Buckets to hold the function code and deploys both functions in the same script.
+
+```python
+from pulumi_gcp import storage, cloudfunctions
+from pulumi import export, asset
+
+bucket = storage.Bucket("bucket")
+
+py_bucket_object = storage.BucketObject(
+    "python-zip",
+    bucket=bucket.name,
+    source=asset.AssetArchive({
+        ".": asset.FileArchive("./pythonfunc")
+    }))
+
+py_function = cloudfunctions.Function(
+    "python-func",
+    source_archive_bucket=bucket.name,
+    runtime="python37",
+    source_archive_object=py_bucket_object.name,
+    entry_point="handler",
+    trigger_http="true",
+    available_memory_mb=128,
+)
+
+export("python_endpoint", py_function.https_trigger_url)
+
+go_bucket_object = storage.BucketObject(
+    "go-zip",
+    bucket=bucket.name,
+    source=asset.AssetArchive({
+        ".": asset.FileArchive("./gofunc")
+    }))
+
+go_function = cloudfunctions.Function(
+    "go-func",
+    source_archive_bucket=bucket.name,
+    runtime="go111",
+    source_archive_object=go_bucket_object.name,
+    entry_point="Handler",
+    trigger_http="true",
+    available_memory_mb=128,
+)
+
+export("go_endpoint", go_function.https_trigger_url)
+```
+
+The [full example](https://github.com/pulumi/examples/tree/master/gcp-ts-serverless-raw) is available on Github.
+
+## Cold Starts
+
+Despite the many advantages of serverless, one of the challenges of serverless is latency, more commonly known as “cold starts.” There are several ways to lessen the impact of cold starts, including:
+
+- warming is the practice of invoking additional workers to provide a pool of workers
+- provisioned concurrency has a set of workers are always warm and dedicated to a specific function
+- dynamic concurrency is similar to provisioned concurrency but the workers can autoscale as needed
+- scheduled profile is warming scheduled during peak use
+- autoscaling based utilization adds workers by measuring utilization
+
+Deploying infrastructure with code lets you implement these [strategies]({{< relref "/blog/aws-lambda-provisioned-concurrency-no-cold-starts" >}}) based on your application requirements. A [worked example](https://github.com/pulumi/examples/tree/master/aws-ts-serverless-raw) is available on Github.
```

Comment:
Is this example related to cold starts?

Reply:
added code sample for provisioned concurrency